src/TinyUPnP.cpp

```diff
@@ -697,25 +720,30 @@ boolean TinyUPnP::getIGDEventURLs(gatewayInfo *deviceInfo) {
 				urlBaseFound = true;
 			}
 		}
+
+		// to support multiple <serviceType> tags
```

Comment:
@GIPdA can you please explain why you needed this?

Reply:
My router's XML file has several <serviceType> tags, and the "right" one isn't the first one. Your implementation didn't worked with several tags. With `service_type_index_start` it will check every tag until it finds one that should work.
It refers to issue #36